package/kedro_viz/api/rest/responses/run_events.py

```diff
@@ -0,0 +1,464 @@
+
+    Args:
+        pipeline_info: Pipeline info object to finalize
+        nodes: Dictionary of node info objects for duration calculation
+    """
+    # Generate unique run_id if using default
+    if pipeline_info.run_id == "default-run-id":
+        pipeline_info.run_id = str(uuid.uuid4())
+
+    # Calculate total pipeline duration
+    node_durations = {node_id: node.duration for node_id, node in nodes.items()}
+    pipeline_info.duration = calculate_pipeline_duration(
+        pipeline_info.start_time,
+        pipeline_info.end_time,
+        node_durations,
+    )
+
+
+def transform_events_to_structured_format(
+    events: list[dict[str, Any]],
+) -> RunStatusAPIResponse:
+    """Convert raw run events into structured API response format.
+
+    Args:
+        events: List of raw event dictionaries
+
+    Returns:
+        Structured API response object containing nodes, datasets, and pipeline info
+    """
+    nodes: dict[str, NodeInfo] = {}
+    datasets: dict[str, DatasetInfo] = {}
+    pipeline = PipelineInfo()
+
+    # Update pipeline metadata first
+    _update_pipeline_info_from_events(events, pipeline)
+
+    # Process events by type
+    for event in events:
+        event_type = event.get("event")
+
+        if event_type == EventType.AFTER_NODE_RUN:
+            _process_node_completion_event(event, nodes)
+        elif event_type == EventType.ON_NODE_ERROR:
+            _process_node_error_event(event, nodes)
+        elif event_type in {
+            EventType.AFTER_DATASET_LOADED,
+            EventType.AFTER_DATASET_SAVED,
+        }:
+            _process_dataset_event(event, datasets)
+        elif event_type == EventType.ON_PIPELINE_ERROR:
+            # Handle pipeline errors that may contain dataset information
+            if "dataset" in event:
+                _process_dataset_error_event(event, datasets, nodes, pipeline)
+
+    # Finalize pipeline information
+    _finalize_pipeline_info(pipeline, nodes)
+
+    return RunStatusAPIResponse(
+        nodes=nodes,
+        datasets=datasets,
+        pipeline=pipeline,
+    )
+
+
+def get_run_status_response() -> RunStatusAPIResponse:
+    """Get run status data for API endpoint in structured format.
+
+    Returns:
+        Structured API response object containing run status data
+
+    Raises:
+        FileNotFoundError: If the run events file cannot be found
+        json.JSONDecodeError: If the run events file contains invalid JSON
+    """
+    try:
+        kedro_project_path = _find_kedro_project(Path.cwd())
+        if not kedro_project_path:
+            logger.warning(
+                "Could not find a Kedro project to load pipeline events file"
+            )
+            return RunStatusAPIResponse()
+
+        pipeline_events_file_path = PIPELINE_EVENT_FULL_PATH
+
+        if not pipeline_events_file_path.exists():
+            logger.warning(f"Run events file {pipeline_events_file_path} not found")
```

Comment:
```suggestion
            logger.warning(f"Run events file {pipeline_events_file_path} not found. It may be due to missing `kedro run`")
```
Not sure of the wording but should we hint the user what might be the reason ?